### PR TITLE
Feat: Added single Video Page and Filters

### PR DESCRIPTION
**Pull request by sainath7878**
Please check the following files:

- [src/components/filters/Filters.jsx](https://github.com/sainath7878/joddtube/compare/filters%26IndividualVideoPage?expand=1#diff-351aac96af7afd9d20494665760830717e22a8a384f4f2972eaabfe6473e24cd)
- [src/hooks/useFetch.jsx](https://github.com/sainath7878/joddtube/compare/filters%26IndividualVideoPage?expand=1#diff-697611df152fff126d2a1d2be956d7e15ba3651aa951be30351b548351f596af)
- [src/hooks/useScrollToTop.jsx](https://github.com/sainath7878/joddtube/compare/filters%26IndividualVideoPage?expand=1#diff-2f30b311d907f9d1a1d58fbbd37a42dea9ca2b29fd4333cd6f32f212d25e399e)
- [src/pages/videoPage/VideoPage.jsx](https://github.com/sainath7878/joddtube/compare/filters%26IndividualVideoPage?expand=1#diff-c3da235c1d2cbf2c1c6bb4d697cc0f3353d4c1eb22992aedb21eafcd9916267e)
- [src/utils/filteredData.js](https://github.com/sainath7878/joddtube/compare/filters%26IndividualVideoPage?expand=1#diff-2f0dc7f7fb61e2d09fc437171b88b46bcfd6846b35236098277f1071281d4e01)
- [src/pages/videoPage/videoPage.module.css](https://github.com/sainath7878/joddtube/compare/filters%26IndividualVideoPage?expand=1#diff-29c0e1dbbdd400731c2148db91bf7c3bf4e9ac2fcd58e77aff8512f9f8a998e0)

### CODE REVIEW

File: src/components/filters/Filters.jsx [hunk 1]
```diff
@@ -1,101 +1,27 @@
...
-  const [filters, setFilters] = useState({
-    all: true,
-    valorant: false,
-    gta: false,
-    csgo: false,
-    bgmi: false,
-  });
+function Filters({ filters, setFilters }) {
+  const { categories } = useFetch("/api/categories");
+  console.log(categories);
```
Comment (BalajiHarinath) [inline src/components/filters/Filters.jsx]: Suggestion: Can remove the console.log

File: src/components/filters/Filters.jsx [hunk 2]
```diff
@@ -1,101 +1,27 @@
...
+                  ? "btn-secondary"
+                  : `${styles.btnSecondaryOutline}`
+              } ${styles.btn}`}
+              key={_id}
+              onClick={() => setFilters(categoryName)}
+            >
+              {categoryName}
+            </button>
+          );
+        })}
```
Comment (BalajiHarinath) [inline src/components/filters/Filters.jsx:23]: Appreciation: good that you have used map for categories.

**Review by BalajiHarinath — APPROVED** [review]
Clean code, could not find any issues. Nice work and UI is awesome.

File: src/pages/videoPage/VideoPage.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                </div>
+              </div>
+            </div>
+            <h2 className={styles.videoDescription}>{video.description}</h2>
+          </div>
+        </div>
+      ) : (
+        <div className="loader">
+          <RotatingLines width="100" strokeColor="#a40ae0" />
+        </div>
```
Comment (anik31) [inline src/pages/videoPage/VideoPage.jsx:75]: appreciation - Liked that you've implemented loader.

File: src/hooks/useScrollToTop.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { useEffect } from "react";
+
+const useScrollToTop = () => {
+  useEffect(() => {
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    });
```
Comment (anik31) [inline src/hooks/useScrollToTop.jsx:8]: appreciation - good that you've added this hook

File: src/hooks/useFetch.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+const useFetch = (api) => {
+  const [response, setResponse] = useState({});
+  useEffect(() => {
+    (async () => {
+      let res;
+      try {
+        res = await axios.get(api);
+        if (res.status === 200) {
+          setResponse(res.data);
+        }
```
Comment (anik31) [inline src/hooks/useFetch.jsx:14]: appreciation - liked that you're using a separate hook for fetch

**Review by anik31 — APPROVED** [review]
Everything looks good. The filter is working fine & single page also looks good.

File: src/components/filters/Filters.jsx (hunk 1 above)
Comment (sainath7878) [reply src/components/filters/Filters.jsx]: Yeah will remove. Thanks for pointing out!

**Pull request merged**